### ISSUE TITLE
Fullscreen toggling on macOS broken

### ISSUE THREAD

**Issue by macsforme** [BZFlag-Dev/bzflag#241]
An issue with fullscreen toggling on macOS was introduced in 1fa9934d.

**Steps to replicate:**
1. Upon launch, switch to windowed mode via F1.
2. Maximize the window via the toolbar.
3. Press F1 to switch to regular fullscreen mode.
4. Press F1 again to switch back to windowed mode.
5. The window content is now offset and partially cut off at the top.

![Screen Shot 2020-01-04 at 1 33 31 AM](https://user-images.githubusercontent.com/4955448/71762873-7b7b4300-2e92-11ea-8fee-b5b76394a5de.png)

**Expected behavior:**

- BZFlag should return to standard windowed mode (non-maximized), but the drawing area should be resized to fit in the new window.

This was discovered while testing whether https://github.com/atupone/bzflag/blob/3mon/src/platform/SDL2Window.cxx#L268-L314 is still required or not. Incidentally, that event discard never seems to fire anymore, so we may be able to eliminate that section once we have a working window creation process again via #228 or #238. However, removing that section has no effect on this issue.

**Comment by atupone** [comment]
I have seen this bug first on 2.4.8

Before that (2.4.6) the resultant window is the same as after toggling the window toolbar 
The 2.4.8 added the little offset on y

**Comment by macsforme** [comment]
After pondering this further, I believe the issue is caused by the game switching into a standard (non-maximized) window upon exiting fullscreen, whereas it had a maximized window matching the fullscreen resolution prior to switching into fullscreen mode. Because a window with the fullscreen resolution does not fit into a standard window with a title bar and other OS GUI elements, the window contents become offset. Furthermore, our window creation mechanism does not currently have a way of telling the game that we got back a different resolution than we requested.

A solution may be to maximize the window again via the SDL 2 API after exiting fullscreen, if we are on macOS and detect that a window is being requested which matches the fullscreen resolution.
